ducc/lib/image.go

```diff
@@ -661,8 +679,15 @@ func (img *Image) downloadLayer(layer da.Layer, token, rootPath string) (toSend
 				LogE(err).Warning("Error in creating the zip to unzip the layer")
 				continue
 			}
-
-			toSend = downloadedLayer{Name: layer.Digest, Path: gread}
+			var path io.ReadCloser
+			path = gread
+			if !skipPodman {
+				var buf bytes.Buffer
+				path = TeeReadCloser(path, &buf)
+				podmaninfo := img.GetPodmanInfo()
+				podmaninfo.LayerReaderMap[layer.Digest] = ReadCloserBuffer{&buf}
```

Comment:
> What happens if we do this operation ALWAYS, no matter if we are storing podman stuff or not? It is a problem? Do we incur in a performance penality?
We can store this information always. This won't cause much performance penalty. Infact, with this we can download layers and convert for docker independently of podman. Will incorporate this.
I'll try using io.Pipe() with TeeReader.

Reply:
Do we need a TeeReader at all?

If I understand the code correctly, we are using the TeeReader to create two stream of data, one is serialized into cvmfs and ingested, the other is used to compute the size and the hash. Am I wrong?

If we just override the Read method (or better, implement), so that it keep track of the size and compute the hash, one piece at the time we can do without the TeeReader.
Basically we are re-implementing it, but we manage only one buffer.

Definitely open to more suggestions on this. We could TeeReader inside the hash function for instance. But then we need some way to keep track of the size...

Reply:
We can use TeeReader to directly write to hash.Hash . We can add functionality to ReadAndHash for calculating the size as well. Something like this: https://play.golang.org/p/sZ-cHoWm2fr

Reply:
Love it! Go for it!

Reply:
Done